pallets/confidential-asset/src/lib.rs

```diff
@@ -36,19 +51,91 @@ pub trait Trait:
 {
 }
 
+/// Wrapper for Ciphertexts that correspond to EncryptedAssetId.
+/// This is needed since `mercat::asset_proofs::elgamal_encryption::CipherText` implements
+/// Encode and Decode, instead of deriving them. As a result, the EncodeLike operator is
+/// not automatically implemented.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, VecU8StrongTyped, Default)]
+pub struct EncryptedAssetIdWrapper(pub Vec<u8>);
+
+/// Wrapper for Ciphertexts that correspond to EncryptedBalance.
+/// This is needed since `mercat::asset_proofs::elgamal_encryption::CipherText` implements
+/// Encode and Decode, instead of deriving them. As a result, the EncodeLike operator is
+/// not automatically implemented.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, VecU8StrongTyped, Default)]
+pub struct EncryptedBalanceWrapper(pub Vec<u8>);
+
+/// A mercat account consists of the public key that is used for encryption purposes and the
+/// encrypted asset id. The encrypted asset id also acts as the unique identifier of this
+/// struct.
+#[derive(Encode, Decode, Clone, Debug, PartialEq, Eq, Default)]
+pub struct MercatAccount {
+    pub encrypted_asset_id: EncryptedAssetIdWrapper,
+    pub encryption_pub_key: EncryptionPubKey,
+}
+
 decl_storage! {
-    trait Store for Module<T: Trait> as Asset {
+    trait Store for Module<T: Trait> as ConfidentialAsset {
+
+        /// Contains the mercat accounts for an identity.
+        pub MercatAccounts get(fn mercat_account): double_map hasher(blake2_128_concat) IdentityId, hasher(blake2_128_concat) EncryptedAssetIdWrapper  => MercatAccount;
+
+        /// Contains the encrypted balance of a mercat account.
+        pub MercatAccountBalance get(fn mercat_account_balance):  double_map hasher(blake2_128_concat) IdentityId, hasher(blake2_128_concat) EncryptedAssetIdWrapper =>EncryptedAssetIdWrapper;
+
+        /// Contains the list of all valid ticker names.
+        /// The process around creating and storing this data will likely change as a result of CRYP-153.
+        pub ValidAssetIds get(fn valid_asset_ids): Vec<AssetId>;
```

Comment:
It is the plain text for the asset?

Reply:
Yes and it will be merged/replaced by the actual ticker names in the next PR.